src/AWS.Distro.OpenTelemetry.AutoInstrumentation/AwsAttributeKeys.cs

```diff
@@ -47,6 +47,12 @@ internal sealed class AwsAttributeKeys
 
     internal static readonly string AttributeAWSS3Bucket = "aws.s3.bucket";
 
+    internal static readonly string AttributeAWSBedrockGuardrailId = "aws.bedrock.guardrail.id";
+    internal static readonly string AttributeAWSBedrockAgentId = "aws.bedrock.agent.id";
+    internal static readonly string AttributeAWSBedrockKnowledgeBaseId = "aws.bedrock.knowledge_base.id";
+    internal static readonly string AttributeAWSBedrockDataSourceId = "aws.bedrock.data_source.id";
+    internal static readonly string AttributeGenAiModelId = "gen_ai.request.model";
+
```

Comment:
I would love these changes to be more config driven rather code for simplicity and extensibility... the way to visualize is to have a simple key/value pair and iterating over it in the code to find key(attributes to look for) and value is servicename and resource type. E:g:-

```json
{
   [
      "service" :  "Kinesis",
      "Name" : "AmazonKinesis, Kinesis",
       "Attributes" :[{"key": "aws.kinesis.stream_name", "Value": "AWS::Kinesis:Stream"}]
  ],
  [
      "service" :  "bedrock",
      "Name" : "Bedrock, Bedrock Agent",
       "Attributes" :[
                {"key": "aws.bedrock.guardrail.id", "value": "AWS::Bedrock::Guardrail"},
                {"key": "gen_ai.request.model", "value": "AWS::Bedrock::Model"}
              //etc
         ]
  ],
}
```

If we can adopt above model both into code and test, this would make easy to add more attributes, services quickly. Eventually we can think that this config as an s3 or appconfig, so without changing code, with config push customer are getting new attributes.